### PR TITLE
feat: Methodology | create the page

### PR DESCRIPTION
**Pull request by Saedbhati**
## Screenshots
image

## Code of Conduct

- [ ] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/Support/blob/main/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

File: src/components/Resources/Methodology/MethodologyElement.jsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import styled from "styled-components";
+export const CategoryHeading = styled.h1`
```
Comment (ArkadiK94) [inline src/components/Resources/Methodology/MethodologyElement.jsx:2]: You don't use it anywhere, so or use it, or delete this file.

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]
@Saedbhati Looks great, I have some comments.
@kabir0x23 I need you opinion on one comment about the path.

File: src/components/Resources/Methodology/MethodologyMain.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { Wrapper } from "../../Dashboard/Profile/ProfileElements";
+import { MethodologyData } from "./MethodlogyData";
+
+const Methodology = () => {
+    function handleClick(id) {
```
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx:6]: @kabir0x23 what do you think?

File: src/App.jsx [hunk 3]
```diff
@@ -146,6 +147,7 @@ const App = () => {
                         <Route path={"/courses/*"} element={<CoursesRoute />} />
 
                         <Route path={"/security/*"} element={<SecurityRoutes />} />
+                        <Route path={"/resources/methodology"} element={<Methodology />} />
```
[the remark this answers is not in the document]
Comment (kabir0x23) [reply src/App.jsx:150]: yes correct, those should be in the /resources path

[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/App.jsx:150]: Ok, so I resolve this comment, and we need to make changes in the other pages as well.

File: src/components/Resources/Methodology/MethodologyMain.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx:6]: https://www.reddit.com/r/reactjs/comments/zvqhku/what_is_the_equivalent_of_documentgetelementbyid/?rdt=41932

**Comment by Saedbhati** [comment]
Ok I will

File: src/components/Resources/Methodology/MethodologyMain.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            document.getElementById(id).style.display = "none";
+        } else {
+            document.getElementById(id).style.display = "block";
+        }
+    }
+    return (
+        <Wrapper>
+            <div></div>
+            <div>
+                {Object.keys(MethodologyData).map((data) => (
```
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx]: Need to update all the places where this parameter is used

File: src/components/Resources/Methodology/MethodologyMain.jsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                        <p
+                            style={{
+                                display: "block",
+                                textAlign: "center",
+                            }}
+                        >
+                            {MethodologyData[data].Description}
+                        </p>
+
+                        {Object.keys(MethodologyData[data].Topics).map((Topics) => (
```
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx]: Need to update all the places where this parameter is used

File: src/components/Resources/Methodology/MethodologyMain.jsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                            <div key={Topics} style={{ width: "100%", margin: "20px" }}>
+                                <h2 style={{ textAlign: "center", fontSize: "25px" }}>{Topics}</h2>
+                                {Object.keys(MethodologyData[data].Topics[Topics]).map((SubTopics) => (
```
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx]: Need to update all the places where this parameter is used

File: src/components/Resources/Methodology/MethodologyMain.jsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+                                        >
+                                            <p>
+                                                {" "}
+                                                <span style={{ fontWeight: "bold" }}>Summary </span>-
+                                                {MethodologyData[data].Topics[Topics][SubTopics].Summary}
+                                            </p>
+                                            <h1 style={{ marginTop: "5px" }}>Goals:</h1>
+                                            <ul style={{ listStyle: "circle", marginLeft: "30px" }}>
+                                                {MethodologyData[data].Topics[Topics][SubTopics].Goals.map(
+                                                    (Goals) => (
```
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx:76]: Need to update all the places where this parameter is used

File: src/components/Resources/Methodology/MethodologyMain.jsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                                            onClick={() => {
+                                                handleClick(SubTopics);
+                                            }}
+                                            style={{
+                                                background: "rgb(19, 19, 19)",
+                                                marginBottom: "0px",
+                                                padding: "15px",
+                                                width: "100%",
+                                                borderBottom: "1px solid white",
+                                                borderTop: "0.5px solid white",
```
[the remark this answers is not in the document]
Comment (ArkadiK94) [reply src/components/Resources/Methodology/MethodologyMain.jsx]: @Saedbhati Did you miss this comment? do you have a different opinion regarding it?

**Review by ArkadiK94 — APPROVED** [review]
@Saedbhati Excellent 💯 Only one minor comment left that you didn't reply about / make changes.

**Comment by Saedbhati** [comment]
@ArkadiK94  sorry i may have overlook it but i have updated the code u check

**Pull request merged**